test/test-assert-subroutine-error-termination.F90

```diff
@@ -47,15 +55,18 @@ program test_assert_subroutine_error_termination
       end if
     end if
   end block
+#else
+#ifdef __LFORTRAN__
+    print *,trim(merge("passes","FAILS ",exit_status/=0)) // " on error-terminating when assertion = .false."
 #else
     block
       integer unit
       open(newunit=unit, file="build/exit_status", status="old")
       read(unit,*) exit_status
-      print *,trim(merge("passes","FAILS ",exit_status/=0)) // " on error-terminating when assertion = .false."
       close(unit)
     end block 
 #endif
+#endif
```

Comment:
This transformation was apparently done to add support for LFortran, but the change has also (inadvertently?) perturbed and broken the behavior for single-image execution on other compilers. 

So for example, the following command no longer produces correct output:
```
 fpm test --compiler gfortran --flag "-DASSERTIONS -DASSERT_MULTI_IMAGE=0"
```

The removed line needs to be restored, and the first two lines should instead be `#elif defined(__LFORTRAN__)`